### ISSUE TITLE
Maven Release with source jar but without OSGI-OPT sources

### ISSUE THREAD

**Issue by swimmesberger** [bndtools/bnd#3091]
We would like to use the release feature of bnd to upload artifacts on our local maven instance but it seems that every bundle is published with the additional (OSGi specific) OSGI-OPT folder additionally to the bunde-source.jar artifact. We would like to use our bundles from maven in other tools which may use a flat maven/gradle based build but for such builds the OSGI-OPT directory is useless and for the engineers which "just want to use" the dependency they would have to take special care to not ship the libraries with the OSGI-OPT directory included.

So long story short: To better fit into the maven universe we would like to a have a way to tell bnd to exclude the OSGI-OPT directory in the "base" jar. Like we can do in \*.bndrun files with the "-executable" instruction (strip=OSGI-OPT/*).

**Comment by bjhargrave** [comment]
Usage questions should be asked in the bndtools-users mail list. See <https://github.com/bndtools/bnd#feedback>.

You want `-sources: false`. https://bnd.bndtools.org/chapters/825-instructions-ref.html

**Comment by swimmesberger** [comment]
I know that instruction but as far as I can tell if I set the sources instruction to false I can't have a bundle-source.jar because the source jar is generated based on the OSGI-OPT/src directory of the base jar. But I want sources and I want a bundle-source.jar but I don't want is a bundle.jar containing a OSGI-OPT/src directory.

**Comment by bjhargrave** [comment]
Again, I would request you to use the mail list for usage type questions.

You can use `-maven-release: sources;path=...` https://bnd.bndtools.org/instructions/maven-release.html

**Comment by swimmesberger** [comment]
Yeah but as soon as you define custom source the javadoc jar does not work because the javadoc generation only considers the source defined in OSGI-OPT directory as you can see here:
https://github.com/bndtools/bnd/blob/6e1af4c3c15bf4f12d9639193a2ab2cffb3bf95a/biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Tool.java#L47

If a custom path for source is specified via instructions the javadoc method still uses the directory of the constructor as you can see here:
https://github.com/bndtools/bnd/blob/6e1af4c3c15bf4f12d9639193a2ab2cffb3bf95a/biz.aQute.repository/src/aQute/bnd/repository/maven/provider/Tool.java#L127

The right way in my opinion would be to consider the path of the source instruction for javadoc generation.

**Comment by bjhargrave** [comment]
> The right way in my opinion would be to consider the path of the source instruction for javadoc generation.

Yes. I can make a fix here.

**Comment by bjhargrave** [comment]
PR #3100 has a fix for this. Note that you will need to set `-maven-release` to properly generate your javadoc so that the javadoc command has access to the buildpath. For example:

`-maven-release: sources;path="${src}",javadoc;-classpath="${project.buildpath}"`
